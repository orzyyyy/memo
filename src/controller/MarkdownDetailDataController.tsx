import React from 'react';
import { useFetchDocumentData } from '../hooks/useFetchDocumentData';
import MarkdownDetail from '../pages/MarkdownDetail';
import { getPathNameFromUrl } from '../utils';

const MarkdownDetailDataController = () => {
  const id: string = getPathNameFromUrl();
  const [data] = useFetchDocumentData(id, 'markdown');
  const screenWidth = document.body.clientWidth;

  return <MarkdownDetail dataSource={data as any} screenSize={screenWidth > 800 ? 'pc' : 'mobile'} />;
};

export default MarkdownDetailDataController;
