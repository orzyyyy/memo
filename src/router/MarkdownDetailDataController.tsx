import React, { Suspense, lazy } from 'react';
import ReactDOM from 'react-dom';
import TohoLoading from '../pages/TohoLoading';

const MarkdownDetailDataController = lazy(() =>
  import('../controller/MarkdownDetailDataController'),
);

ReactDOM.render(
  <Suspense fallback={<TohoLoading />}>
    <MarkdownDetailDataController />
  </Suspense>,
  document.getElementById('root'),
);
