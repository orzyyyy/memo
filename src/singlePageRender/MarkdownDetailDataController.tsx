import React from 'react';
import ReactDOM from 'react-dom';
import MarkdownDetailDataController from '../controller/MarkdownDetailDataController';

ReactDOM.render(
  <MarkdownDetailDataController />,
  document.getElementById('root'),
);
