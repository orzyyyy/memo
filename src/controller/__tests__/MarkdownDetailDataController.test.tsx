import React from 'react';
import { mount } from 'enzyme';
import MarkdownDetailDataController from '../MarkdownDetailDataController';
import fetchMock from 'fetch-mock';

describe('MarkdownDetailDataController', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  beforeEach(() => {
    fetchMock.mock('/api/memo/document/update', {});
    fetchMock.mock('/markdown//.md', { test: 1 });
  });

  afterAll(() => {
    errorSpy.mockRestore();
  });

  afterEach(() => {
    fetchMock.restore();
  });

  it('handleOnSave', async () => {
    const wrapper = mount(<MarkdownDetailDataController />);
    await wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
});
