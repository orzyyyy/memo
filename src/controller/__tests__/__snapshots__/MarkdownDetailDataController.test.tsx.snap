// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownDetailDataController handleOnSave 1`] = `
<MarkdownDetailDataController>
  <_default>
    <div
      className="MarkdownDetail markdown-body"
    >
      <div
        className="wrapper"
      >
        <div
          className="content"
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </div>
    </div>
  </_default>
</MarkdownDetailDataController>
`;
